pints/_mcmc/_slice_rank_shrinking.py

```diff
@@ -0,0 +1,270 @@
+# -*- coding: utf-8 -*-
+#
+# Slice Sampling - Covariance Adaptive: Rank Shrinking
+#
+# This file is part of PINTS.
+#  Copyright (c) 2017-2019, University of Oxford.
+#  For licensing information, see the LICENSE file distributed with the PINTS
+#  software package.
+#
+from __future__ import absolute_import, division
+from __future__ import print_function, unicode_literals
+import pints
+import numpy as np
+
+
+class SliceRankShrinkingMCMC(pints.SingleChainMCMC):
+    r"""
+    Implements Covariance-Adaptive Slice Sampling by Rank Shrinking,
+    as described in [1]_. This is an adaptive multivariate method which uses
```

Comment:
Small comment: _In general_, docstrings should be a one-liner, then optionally a new paragraph with more info. So 2nd sentence here would be better suited as 1st sentence of following paragraph

Reply:
Errm not sure we've done this elsewhere, so can we leave for now?

Reply:
I've done it myself for now. We've been quite good at doing this consistently! It's in the guidelines, but I've added some more links to make this clearer in https://github.com/pints-team/pints/pull/1075

As guidelines go, this is a really good one! It means tools (e.g. IDEs) can show a single line description of any class/method/function